cdap-proto/src/main/java/co/cask/cdap/proto/codec/AbstractSpecificationCodec.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
```

Comment:
Copied from cdap-app-fabric. Couldn't simply move b/c it's also used for other codecs, but I didn't want to move all the codecs over since I don't know what the impact is.